### ISSUE TITLE
getCurrentPosition crashes internally

### ISSUE THREAD

**Issue by livotov** [transistorsoft/flutter_background_geolocation#80]
* Plugin version: 1.0.8
* Platform: Android
* OS version: 9
* Device manufacturer / model: Pixel 2
* Flutter info (`flutter doctor`):

```
[✓] Flutter (Channel stable, v1.5.4-hotfix.2, on Mac OS X 10.14.3 18D109, locale en-BG)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
[✓] Android Studio (version 3.4)
[✓] IntelliJ IDEA Ultimate Edition (version 2018.3.5)
[✓] Connected device (2 available)

• No issues found!
```

* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
bg.BackgroundGeolocation.ready(bg.Config(
        desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
        distanceFilter: 10.0,
        stopOnTerminate: false,
        startOnBoot: true,
        debug: true,
        enableHeadless: true,
        logLevel: bg.Config.LOG_LEVEL_INFO,
        reset: true
    )).then((bg.State state) {
      print('[ready] success: $state');
      if (!state.enabled) {
        bg.BackgroundGeolocation.start();
      }
    });
```

## Expected Behavior
getCurrentPosition's future should either return location or call error callback
## Actual Behavior
it crashes internally (can't even catch the crash)

## Steps to Reproduce
```dart
bg.BackgroundGeolocation.getCurrentPosition(
          timeout: 30,
          maximumAge: 600000,
          desiredAccuracy: 40,
          samples: 3
      ).then((location) => {}).catchError((err) => {});
```

## Debug logs
``` <!-- syntax-highligting:  DO NOT REMOVE -->
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): Failed to handle method call
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at com.transistorsoft.flutter.backgroundgeolocation.FLTBackgroundGeolocationPlugin.getCurrentPosition(FLTBackgroundGeolocationPlugin.java:392)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at com.transistorsoft.flutter.backgroundgeolocation.FLTBackgroundGeolocationPlugin.onMethodCall(FLTBackgroundGeolocationPlugin.java:198)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:201)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:88)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:219)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at android.os.MessageQueue.next(MessageQueue.java:326)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at android.os.Looper.loop(Looper.java:160)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at android.app.ActivityThread.main(ActivityThread.java:6718)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/MethodChannel#com.transistorsoft/flutter_background_geolocation/methods(12180): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
E/flutter (12180): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: FormatException: Invalid radix-10 number (at character 1)
E/flutter (12180): error
E/flutter (12180): ^
E/flutter (12180): 
E/flutter (12180): #0      int._throwFormatException (dart:core-patch/integers_patch.dart:131:5)
E/flutter (12180): #1      int._parseRadix (dart:core-patch/integers_patch.dart:142:16)
E/flutter (12180): #2      int._parse (dart:core-patch/integers_patch.dart:100:12)
E/flutter (12180): #3      int.parse (dart:core-patch/integers_patch.dart:63:12)
E/flutter (12180): #4      new LocationError (package:flutter_background_geolocation/models/location.dart:166:16)
E/flutter (12180): #5      BackgroundGeolocation.getCurrentPosition.<anonymous closure> (package:flutter_background_geolocation/background_geolocation.dart:491:35)
E/flutter (12180): #6      _rootRunUnary (dart:async/zone.dart:1132:38)
E/flutter (12180): #7      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter (12180): #8      _FutureListener.handleError (dart:async/future_impl.dart:144:20)
E/flutter (12180): #9      Future._propagateToListeners.handleError (dart:async/future_impl.dart:651:47)
E/flutter (12180): #10     Future._propagateToListeners (dart:async/future_impl.dart:672:24)
E/flutter (12180): #11     Future._completeError (dart:async/future_impl.dart:491:5)
E/flutter (12180): #12     _SyncCompleter._completeError (dart:async/future_impl.dart:55:12)
E/flutter (12180): #13     _Completer.completeError (dart:async/future_impl.dart:27:5)
E/flutter (12180): #14     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:40:18)
E/flutter (12180): #15     MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart)
E/flutter (12180): <asynchronous suspension>
E/flutter (12180): #16     BackgroundGeolocation.getCurrentPosition (package:flutter_background_geolocation/background_geolocation.dart:487:10)
E/flutter (12180): <asynchronous suspension>
E/flutter (12180): #17     OrderDetailsScreenState._startStatusChange (package:papuio/order.dart:83:32)
E/flutter (12180): #18     OrderDetailsScreenState._onFinishOrderUpdate (package:papuio/order.dart:78:5)
E/flutter (12180): #19     OrderDetailsScreenState._onChangeStatus.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:papuio/order.dart:121:50)
E/flutter (12180): #20     _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:511:14)
E/flutter (12180): #21     _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:566:30)
E/flutter (12180): #22     GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:166:24)
E/flutter (12180): #23     TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:240:9)
E/flutter (12180): #24     TapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:211:7)
E/flutter (12180): #25     GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:156:27)
E/flutter (12180): #26     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:225:20)
E/flutter (12180): #27     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:199:22)
E/flutter (12180): #28     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:156:7)
E/flutter (12180): #29     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:102:7)
E/flutter (12180): #30     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:86:7)
E/flutter (12180): #31     _rootRunUnary (dart:async/zone.dart:1136:13)
E/flutter (12180): #32     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter (12180): #33     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
E/flutter (12180): #34     _invoke1 (dart:ui/hooks.dart:233:10)
E/flutter (12180): #35     _dispatchPointerDataPacket (dart:ui/hooks.dart:154:5)
E/flutter (12180): 
```

</details>

**Comment by livotov** [comment]
It looks like there is a bug in FLTBackgroundGeolocationPlugin.java:392
I've added proper casting and it works now.

**Comment by christocracy** [comment]
Yes, `maximumAge` is tricky.  iOS wants `double`, Android wants `long`, the Dart API provides `int` to both.
```java
if (options.containsKey("maximumAge"))      {
 builder.setMaximumAge(((Integer) options.get("maximumAge")).longValue()); 
}
```

**Comment by christocracy** [comment]
Fixed in `1.0.9`
